### PR TITLE
Upload test log as artifact

### PR DESCRIPTION
**Pull request by clarenceli-msft**
## Description

Upload the tinylicious log as artifact automatically

### CODE REVIEW

File: tools/pipelines/templates/build-npm-package.yml [hunk 1]
```diff
@@ -305,6 +314,8 @@ stages:
                 mkdir $(Build.ArtifactStagingDirectory)/pack/
                 mkdir $(Build.ArtifactStagingDirectory)/pack/scoped/
                 mkdir $(Build.ArtifactStagingDirectory)/test-files/
+                mkdir $(Build.ArtifactStagingDirectory)/log/
+                mv ${{ parameters.buildDirectory }}/packages/test/test-end-to-end-tests/tinylicious.log $(Build.ArtifactStagingDirectory)/log/
```
Comment (clarenceli-msft) [inline tools/pipelines/templates/build-npm-package.yml:318]: @curtisman @anthony-murphy  Do you think we need to wrap this relative path as an additional path variable?

File: tools/pipelines/templates/build-npm-package.yml [hunk 2]
```diff
@@ -259,6 +259,15 @@ stages:
...
           - ${{ if eq(variables['testCoverage'], true) }}:
+            - task: Npm@1
+              displayName: npm run ci:test:coverage
+              inputs:
+                command: 'custom'
+                workingDir: ${{ parameters.buildDirectory }}
+                customCommand: 'run ci:test:coverage'
+            # Some webpacked file using externals introduce file name with quotes in them
+            # and Istanbul's cobertura reporter doesn't escape them causing parse error when we publish
+            # A quick fix to patch the file with sed. (See https://github.com/bcoe/c8/issues/302)
```
Comment (scarlettjlee) [inline tools/pipelines/templates/build-npm-package.yml:270]: Remove duplicated comment. Either leave at top or move next to the task where sed is used.

Comment (clarenceli-msft) [reply tools/pipelines/templates/build-npm-package.yml]: > Remove duplicated comment. Either leave at top or move next to the task where sed is used.

I do not why the duplicated comment exists after I checked the commit history. I just removed one

File: tools/pipelines/templates/build-npm-package.yml [hunk 3]
```diff
@@ -259,6 +259,12 @@ stages:
           # and Istanbul's cobertura reporter doesn't escape them causing parse error when we publish
           # A quick fix to patch the file with sed. (See https://github.com/bcoe/c8/issues/302)
           - ${{ if eq(variables['testCoverage'], true) }}:
+            - task: Npm@1
+              displayName: npm run ci:test:coverage
+              inputs:
+                command: 'custom'
+                workingDir: ${{ parameters.buildDirectory }}
+                customCommand: 'run ci:test:coverage'
```
Comment (scarlettjlee) [inline tools/pipelines/templates/build-npm-package.yml:267]: I'm not sure what this task is for. Does it cause the tests to run again?

File: tools/pipelines/templates/build-npm-package.yml [hunk 4]
```diff
@@ -345,6 +353,13 @@ stages:
               ArtifactName: 'test-files'
               publishLocation: 'Container'
 
+          - task: PublishBuildArtifacts@1
+            displayName: Publish Artifact - Tinylicious Log
```
Comment (alexvy86) [inline tools/pipelines/templates/build-npm-package.yml:357]: I could envision this new artifact eventually containing other logs as well. We can change the display name at that time though, just pointing it out. I do like that the temp path is `/log` and the artifact name is `log`. If anything I'd use `logs` for both, but just a nit.

File: tools/pipelines/templates/build-npm-package.yml (hunk 3 above)
Comment (clarenceli-msft) [reply tools/pipelines/templates/build-npm-package.yml]: > I'm not sure what this task is for. Does it cause the tests to run again?

I have removed this part, it seems I kept that when I resolved the conflict with Curtis' latest update

File: tools/pipelines/templates/build-npm-package.yml (hunk 4 above)
Comment (clarenceli-msft) [reply tools/pipelines/templates/build-npm-package.yml:351]: Hey Alex, that is a good point and I have changed the name of directory to `/logs`. Currently for the artifact file is just the `tinylicious.log` (with a new location), I suppose we may include other logs in the future, but they will be different artifacts. So I just changed the artifact name `log` to `tinyliciousLog`. May I know what do you think?

**Review by scarlettjlee — APPROVED** [review]

**Pull request merged**